module_1/src/MovieForm/MovieForm.js

```diff
@@ -1,115 +1,138 @@
-import React, {useState} from "react";
-import './MovieForm.css';
+import React from "react";
+import { useNavigate } from 'react-router-dom';
 import GenreSelect from "../GenreSelect/GenreSelect";
-import { faCalendar } from '@fortawesome/free-solid-svg-icons'
-import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import './MovieForm.css';
 
 
 function MovieForm(props) {
-    const calendarIcon = <FontAwesomeIcon icon={faCalendar} />;
     const genres = ['Crime', 'Documentary', 'Horror', 'Comedy'];
+    let selectedGenres = props.movieData? props.movieData.genres : [];
 
-    const [genre, setGenre] = useState([props.movieData.genre]);
-
-    const [title, setTitle] = useState(props.movieData.title);
-    const handleTitleChange = ((e)=> {
-        setTitle(e.target.value)
-    });
-
-    const [url, setUrl] = useState(props.movieData.url);
-    const handleUrlChange = ((e)=> {
-        setUrl(e.target.value)
-    });
-
-    const [rating, setRating] = useState(props.movieData.rating);
-    const handleRatingChange = ((e)=> {
-        setRating(e.target.value)
-    });
-
-    const [releaseDate, setReleaseDate] = useState(props.movieData.releaseDate);
-    const handleReleaseDateChange = ((e)=> {
-        setReleaseDate(e.target.value)
-    });
-
-    const [runtime, setRuntime] = useState(props.movieData.runtime);
-    const handleRuntimeChange = ((e)=> {
-        setRuntime(e.target.value)
-    });
+    const navigate = useNavigate();
+    const AddMovieSchema = Yup.object().shape({
```

Comment:
I think you don't need to keep the schema inside your component. It will be re-created in every render. Better to keep it outside. It does not depend on any props or inner variables, as far as I see